experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+
+        print(scan_combos)
+
+        combos = list_tests.slice_combos_sorter(scan_combos, self.slice_ids)
+
+        return combos
+
+    def check_slice_minimum_requirements(self, exp_slice):
+        """
+        Check for the minimum requirements of the slice. The following keys are always required:
+        "pulse_sequence", "mpinc", "pulse_len", "nrang", "frang", "intt", "intn", "beam_angle",
+        and "beam_order". This function may modify these keys. Ensure the values make sense.
+        :param exp_slice: 
+        :return: 
+        """
+
+        # TODO: add checks for values that make sense, not just check for types
+
+        try:
+            assert 'pulse_sequence' in exp_slice.keys()
+            assert isinstance(exp_slice['pulse_sequence'], list)
+            for element in exp_slice['pulse_sequence']:
+                assert isinstance(element, int)
+        except AssertionError:
+            errmsg = "Slice must specify pulse_sequence that must be a list of integers"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'mpinc' in exp_slice.keys()
+            assert isinstance(exp_slice['mpinc'], int)
+        except AssertionError:
+            errmsg = "Slice must specify mpinc that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:  # TODO may want to add a field for range_gate which could set this param.
+            assert 'pulse_len' in exp_slice.keys()
+            assert isinstance(exp_slice['pulse_len'], int)
+        except AssertionError:
+            errmsg = "Slice must specify pulse_len that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'nrang' in exp_slice.keys()
+            assert isinstance(exp_slice['nrang'], int)
+        except AssertionError:
+            errmsg = "Slice must specify nrang that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'frang' in exp_slice.keys()
+            assert isinstance(exp_slice['frang'], int)
+        except AssertionError:
+            errmsg = "Slice must specify frang that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'intt' in exp_slice.keys()
+        except AssertionError:
+            try:
+                assert 'intn' in exp_slice.keys()
+            except AssertionError:
+                errmsg = "Slice must specify either an intn or intt"
+                raise ExperimentException(errmsg, exp_slice)
+            else:
+                try:
+                    assert isinstance(exp_slice['intn'], int)
+                except AssertionError:
+                    errmsg = "intn must be an integer"
+                    raise ExperimentException(errmsg, exp_slice)
+        else:
+            try:
+                assert isinstance(exp_slice['intt'], int)
```

Comment:
I believe we can use floats, is intt in seconds or other unit? In ROS I believe that there's currently a seconds and a nanoseconds variable, both integers.